tsconfig.json

```diff
@@ -1,18 +1,32 @@
 {
   "compilerOptions": {
-    "module": "es2020",
-    "moduleResolution": "node",
-    "lib": ["es2015", "dom"],
-    "baseUrl": ".",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
+    "lib": [
+      "es2022"
```

Comment:
What would you think about extending https://www.npmjs.com/package/@tsconfig/node18?

Reply:
I "extended" (copied) the version of https://github.com/total-typescript/tsconfig/blob/main/tsc/no-dom/library.json. It's very similar, it's another opinionated preset based on Matt Pococks [TSConfig cheat sheet](https://www.totaltypescript.com/tsconfig-cheat-sheet).

Most important settings are (in my opinion):
- `module` / `moduleResolution` should be either `node16` or `nodenext` when using `tsc` to compile into ESM. This is the case for both versions, although one sticks to `node16` and the other chooses `nodenext`. I am not sure if there is any difference between the two at this point. I see most projects use `nodenext`, but I wouldn't be surprised if they're just aliases with only semantic differences.
-  `target` should be compatible with Node 18 or higher, in our case. This is the case for both versions.
- for libraries its important to use `"declaration": true` so we emit type-definitions as well (this important now, since we now only have `.ts` files in our source), and also `"sourceMap": true` for the same reason.
- the other compiler flags are just additional (opinionated) tweaks on strictness and other optimizations.
- the `lib` can indeed be upgraded to `es2023` for Node 18+